### PR TITLE
#166214115 user should reset password

### PR DESCRIPTION
**Pull request by danndav**
***What does this PR do?***
Enable a registered user to reset his forgotten password via an email notification

Description of Task to be completed?
- forget password implementation
- receive an email for password reset

***How should this be manually tested?***
clone the repository
npm install
add a .env file with the appropiate content as specified in the .env_sample file
sign up as specified in the swagger docs
go to route localhost:3000/api/v1

/passwordreset and insert your registered mail in the body of the request
An email containing link to reset your password will be delivered to the registered mail 
click on the link to reset your password

Any background context you want to provide?
Screenshot from 2019-09-03 12-27-37
***What are the relevant pivotal tracker stories?***
[#166214115](https://www.pivotaltracker.com/story/show/167727456l)

### CODE REVIEW

**Review by parkerthegeniuschild — CHANGES_REQUESTED** [review]
[x] Rename your PR title according to spec.
---
image

**Comment by danndav** [comment]
> ## [x] Rename your PR title according to spec.
> image

alright thanks, i have done that

File: src/controllers/user.controller.js [hunk 1]
```diff
@@ -70,4 +70,58 @@ export default class UserController {
...
+  static async resetpasswordEmail(req, res) {
+    let userToken;
+    let payload;
+    const userData = UserUtils.getUserSignupData(req.body);
+
+    await User.findOne({ attributes: ['first_name', 'email', 'id'], where: { email: userData.email } })
+      .then((data) => {
+        payload = data;
+        userToken = JWTService.generatePasswordToken(data);
+        // res.json(data);
```
Comment (O-Osahon) [inline src/controllers/user.controller.js]: NOTHING MUCH: please always try to remove commented-out code before pushing

File: src/routes/api/auth.router.js [hunk 2]
```diff
@@ -2,11 +2,13 @@ import express from 'express';
...
 import UserMiddleware from '../../middlewares/user.middleware';
 
+
 const router = express.Router();
 
 /* Users Routes Here */
 router.post('/auth/signup', UserController.signup);
 router.post('/auth/signin', ...UserMiddleware.validateSigninFields(),
   UserController.signin);
+router.post('/passwordreset', UserController.resetpasswordEmail);
```
Comment (O-Osahon) [inline src/routes/api/auth.router.js]: This url could use an hyphen like so:
` /password-reset`

**Review by O-Osahon — CHANGES_REQUESTED** [review]
Good job bro. Please make those changes while I test the app

**Comment by parkerthegeniuschild** [comment]
So I tried testing again today.

[x] There is no `.env_sample` file to use as a guide.
---
image
---
[x] Resolve conflicts
---
image
---
[x] Hypenate multi-worded routes
---
image
---
[x]

**Comment by parkerthegeniuschild** [comment]
[x] I am going to approve this; although `the email never came.`
---

**Review by parkerthegeniuschild — APPROVED** [review]

**Review by O-Osahon — APPROVED** [review]
Great job!

**Pull request merged**